txpipe/source_selector.py

```diff
@@ -74,6 +73,9 @@ def run(self):
         # Suppress some warnings from numpy that are not relevant
         original_warning_settings = np.seterr(all='ignore')  
 
+        # Are we using a metacal or lensfit catalog?
```

Comment:
do we want the code to decide itself what kind of shear catalog it is instead of setting it as a config option? could you explain what is the rational?

Reply:
Right, so now we examine the file itself to check what kind of data it contains, or whether it has a metadata entry saying what it is.  That way you don't have to keep manually specifying.